custom_components/bhyve/switch.py

```diff
@@ -515,6 +525,65 @@ def is_on(self):
         """Return the status of the sensor."""
         return self._is_on
 
+    async def set_smart_watering_soil_moisture(self, percentage):
+        """Set the soil moisture percentage for the zone."""
+        if self._attrs[ATTR_SMART_WATERING_ENABLED]:
+            landscape = None
+            try:
+                landscape = await self._bhyve.get_landscape(self._device_id, self._zone_id)
+            
+            except BHyveError as err:
+                _LOGGER.warning(f"Unable to retreive current soil data for {self.name}: {err}")
+            
+            if landscape is not None:
+                _LOGGER.debug("Landscape data %s", landscape)
+
+                # Define the landscape update json payload
```

Comment:
Have you tested that every attribute listed here is required for the update?

Reply:
I did, it can reduce it to the following and it does change the value.
```
  landscape_update = {
      "current_water_level": 0,
      "device_id": "",
      "id": "",
      "station": 0
  }
```
I wasn't sure what the implications might be, the app sends the full list even though you only alter 'current_water_level'.